sys/riscv/boot.c

```diff
@@ -63,21 +65,30 @@
  */
 #define KL_LOG KL_INIT
 #include <sys/fdt.h>
+#include <sys/kasan.h>
 #include <sys/klog.h>
 #include <sys/mimiker.h>
 #include <sys/pcpu.h>
 #include <sys/pmap.h>
 #include <riscv/abi.h>
 #include <riscv/cpufunc.h>
 #include <riscv/pmap.h>
-#include <riscv/vm_param.h>
 
 #define KERNEL_VIRT_IMG_END align((vaddr_t)__ebss, PAGESIZE)
 #define KERNEL_PHYS_IMG_END align(RISCV_PHYSADDR(__ebss), PAGESIZE)
-#define KERNEL_PHYS_END (KERNEL_PHYS_IMG_END + BOOTMEM_SIZE)
+
+#if KASAN
+#define BOOT_KASAN_SANITIZED_SIZE                                              \
+  roundup2(roundup2(KERNEL_VIRT_IMG_END, GROWKERNEL_STRIDE) +                  \
+             (VM_PAGE_PDS * GROWKERNEL_STRIDE) - KASAN_SANITIZED_START,        \
+           PAGESIZE * KASAN_SHADOW_SCALE_SIZE)
+
+#define BOOT_KASAN_SHADOW_SIZE                                                 \
+  (BOOT_KASAN_SANITIZED_SIZE / KASAN_SHADOW_SCALE_SIZE)
+#endif /* !KASAN */
```

Comment:
Is that condition needed here? Could you just create unused definition?

Reply:
Good point. This is the same remark as @cahirwpz has regarding the FPU- and SUM-related defines. Our preferred solution is not to enclose defines unnecessarily (even if they are unused with a specific configuration).